lib/internal/fs/utils.js

```diff
@@ -709,14 +709,30 @@ const validateOffsetLengthWrite = hideStackFrames(
   },
 );
 
-const validatePath = hideStackFrames((path, propName = 'path') => {
+const validatePath = hideStackFrames((path, propName = 'path', isFile) => {
   if (typeof path !== 'string' && !isUint8Array(path)) {
     throw new ERR_INVALID_ARG_TYPE.HideStackFramesError(propName, ['string', 'Buffer', 'URL'], path);
   }
 
   const pathIsString = typeof path === 'string';
   const pathIsUint8Array = isUint8Array(path);
 
+  if (isFile) {
+    const lastCharacter = path[path.length - 1];
+    if (
+      lastCharacter === '/' || lastCharacter === 47 ||
+      (isWindows && (lastCharacter === '\\' || lastCharacter === 92))
+    ) {
+      throw new ERR_FS_EISDIR({
+        code: 'ERR_FS_EISDIR',
+        message: 'is a directory',
+        path,
+        syscall: 'validatePath',
```

Comment:
```suggestion
```

Reply:
This is not a syscall.

Reply:
I've applied the fix you suggested, but as you can see the CI doesn't accept throwing an error without a syscall. I didn't get this on Windows, only Linux.
This function is being called from a number of functions that contain different syscalls. Do you have any idea what I can write here?

Reply:
I think this `syscall` should be the name of the `fs` caller function. Maybe passing it as an argument would be the correct call here? For example, if `fs.cpSync()` is throwing this error, syscall should be `cp`. IMHO, Ideally we should throw this error from C++.

Reply:
Thanks for the review. Fixed.